src/prodbg/main/src/windows.rs

```diff
@@ -258,56 +256,66 @@ impl Window {
             State::Default => {
-                if let Some(h) = self.ws.is_hovering_sizer(mouse_pos) {
-                    if h.1 == Direction::Vertical {
-                    	self.win.set_cursor_style(CursorStyle::ResizeLeftRight);
-                    } else {
-                    	self.win.set_cursor_style(CursorStyle::ResizeUpDown);
-                    }
-
+                if let Some(target) = self.ws.get_drag_target_at_pos(mouse_pos) {
+                    cursor = match target {
+                        DragTarget::Dock(_) => CursorStyle::OpenHand,
+                        DragTarget::SplitSizer(_, direction) => match direction {
+                            Direction::Vertical => CursorStyle::ResizeLeftRight,
+                            Direction::Horizontal => CursorStyle::ResizeUpDown
+                        }
+                    };
                     if self.win.get_mouse_down(MouseButton::Left) {
-                        self.mouse_state.handle = Some(h);
-                        self.mouse_state.state = State::DraggingSlider;
+                        next_state = Some(State::Dragging(target));
                     }
                 } else {
-					self.win.set_cursor_style(CursorStyle::Arrow);
+                    cursor = CursorStyle::Arrow;
                 }
             },
 
-            State::DraggingSlider => {
-                let pm = self.mouse_state.prev_mouse;
-                let delta = (pm.0 - mouse_pos.0, pm.1 - mouse_pos.1);
+            State::Dragging(DragTarget::SplitSizer(handle, direction)) => {
+                if self.win.get_mouse_down(MouseButton::Left) {
+                    cursor = match direction {
+                        Direction::Vertical => CursorStyle::ResizeLeftRight,
+                        Direction::Horizontal => CursorStyle::ResizeUpDown
+                    };
+                    let pm = self.mouse_state.prev_mouse;
+                    let delta = (pm.0 - mouse_pos.0, pm.1 - mouse_pos.1);
+                    self.ws.drag_sizer(handle, delta);
+                } else {
+                    next_state = Some(State::Default);
+                    cursor = CursorStyle::Arrow;
+                }
+            },
 
+            State::Dragging(DragTarget::Dock(handle)) => {
+                let drop_target = self.ws.get_drop_target_at_pos(mouse_pos);
                 if self.win.get_mouse_down(MouseButton::Left) {
-                    let t = self.mouse_state.handle.unwrap();
-                    if t.1 == Direction::Vertical {
-                    	self.win.set_cursor_style(CursorStyle::ResizeLeftRight);
-                    } else {
-                    	self.win.set_cursor_style(CursorStyle::ResizeUpDown);
+                    cursor = match drop_target {
+                        Some(DropTarget::Dock(target)) if target != handle => CursorStyle::OpenHand,
+                        // TODO: make sure this cursor style works. Did not work with minifb 0.8.0
```

Comment:
It is a matter of taste. The only idea to use different cursors here is to
show that area is droppable. When header is hovered, opened hand is used.
When header is dragged, hand is changed to closed hand. When header is
dragged over droppable area (other header), cursor is changed to opened
hand again.

2016-06-28 1:10 GMT+08:00 Daniel Collin <notifications@github.com>:

> In src/prodbg/main/src/windows.rs
> <https://github.com/emoon/ProDBG/pull/205#discussion_r68616315>:
>
> >                  if self.win.get_mouse_down(MouseButton::Left) {
> > -                    let t = self.mouse_state.handle.unwrap();
> > -                    if t.1 == Direction::Vertical {
> > -                    	self.win.set_cursor_style(CursorStyle::ResizeLeftRight);
> > -                    } else {
> > -                    	self.win.set_cursor_style(CursorStyle::ResizeUpDown);
> > +                    cursor = match drop_target {
> > +                        Some(DropTarget::Dock(target)) if target != handle => CursorStyle::OpenHand,
> > +                        // TODO: make sure this cursor style works. Did not work with minifb 0.8.0
>
> Ok. I think I just use "hand2"
> https://github.com/emoon/rust_minifb/blob/master/src/native/x11/X11MiniFB.c#L88
> for both Open/Closed because I couldn't find a standard Open/Closed Hand on
> there.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/emoon/ProDBG/pull/205/files/d46605764f3ac72ffdff6cf828726859576b057f#r68616315>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AANveObOjJOtlAQ36bXWT_MRc4tbBORCks5qQAQhgaJpZM4I_ICN>
> .
>